src/main/java/com/commercetools/pingpong/service/impl/GameServiceImpl.java

```diff
@@ -4,9 +4,11 @@
 import com.commercetools.pingpong.model.Message;
 import com.commercetools.pingpong.model.Player;
 import com.commercetools.pingpong.service.GameService;
+import org.springframework.context.annotation.Primary;
 import org.springframework.stereotype.Component;
 
 @Component
+@Primary
```

Comment:
When we start a ping pong game, the template gets all required information from the *gameService* **interface**. For now, there is just one implementation of the interface so spring boot knows which implementation he has to use, without specifying it concretely.

As you can see, the name of the implementation class is not specified...just the *gameService* interface
https://github.com/TimonRey/PingPongApp/pull/20/commits/9ecc6da1f1279412785abf30fa9b7635296bfedc#diff-aa379526cf99272c6bd56552f614c180R25

 But if we want to implement new **features**, like playing *two vs two*, the easiest way to implement that would be to create a second implementation to specify the required values. If the implementation class is not specified (like in this case), the implementation will always be the *gameServiceImpl* because of the ***@primary***